### PR TITLE
Update TiKV dependency to 3.0.0. 

### PR DESCRIPTION
**Pull request by kennytm**
## What have you changed? (mandatory)

* Update TiKV dependency to 3.0.0 (3.0 GA). 
* Backport the two PRs accepted into 3.0.0 affecting the copy of importer over there:
    * tikv/tikv#4502
    * tikv/tikv#4936 

## What are the type of the changes? (mandatory)

- Improvement (change which is an improvement to an existing feature)

## How has this PR been tested? (mandatory)

cargo check passes

## Does this PR affect TiDB Lightning? (mandatory)

No

### CODE REVIEW

**Comment by kennytm** [comment]
PTAL @leoppro @lance6716

**Review by lance6716 — COMMENTED** [review]
LGTM.

though I'm still learning and don't fully understand logic in stream.rs, I did my best to check that logic keeps same with previous version plus a RocksDB Error.

**Comment by zier-one** [comment]
LGTM

File: src/bin/tikv-importer.rs [hunk 1]
```diff
@@ -3,10 +3,7 @@
...
 #![feature(proc_macro_hygiene)]
 
-#[macro_use(
-    slog_crit,
-    slog_info,
-)]
+#[macro_use(slog_crit, slog_info)]
 extern crate slog;
 #[macro_use]
 extern crate slog_global;
```
Comment (zier-one) [inline src/bin/tikv-importer.rs:9]: do we need `extern crate` in rust 2018 edition?

Comment (kennytm) [reply src/bin/tikv-importer.rs:9]: We don't. It's needed just to import all macros everywhere via `#[macro_rules]`. Rewriting these isn't high priority.

**Comment by kennytm** [comment]
@leoppro @lance6716 could either one of you approve the PR? Needs ≥1 approval to merge. Thanks.

**Review by zier-one — APPROVED** [review]

**Comment by zier-one** [comment]
> @leoppro @lance6716 could either one of you approve the PR? Needs ≥1 approval to merge. Thanks.

I am sorry, I forgot it

**Pull request merged**